web/grpcservice/autograder_service.go

```diff
@@ -118,6 +119,10 @@ func (s *AutograderService) GetCourses(ctx context.Context, in *pb.Void) (*pb.Co
 
 // GetCoursesWithEnrollment returns all courses for the user where user enrollment is of a certain type
 func (s *AutograderService) GetCoursesWithEnrollment(ctx context.Context, in *pb.RecordRequest) (*pb.Courses, error) {
+	if in.ID < 1 {
```

Comment:
Would it make sense to add a validation function for the `RecordRequest` type in validation.go?
Also note that there is benefit to using the `GetID()` (and similar for all other field accessor functions), since allows the `in` (or request) object itself to be `nil` and the function would still work; it would return `nil` or `0` depending on the return type.

Reply:
Commented before checking the next commit, which does exactly the first thing with the validation function... (so ignore that part of the comment...)

Reply:
Then it is probably safer to use getters instead of dot notation everywhere in the code?

Reply:
Yes, it is safer. But it is a tradeoff between safety vs readability. I mention it mainly so that you are aware that the methods does this `nil` check, so that, if you also find the need to do a `nil` check yourself, then that would be unnecessary... and you should use the provide accessor methods. (I'm not suggesting that we should change every instance of using the field variables directly... at least not yet.)